### PR TITLE
Fix rest documentation for changes to spring-rest-docs

### LINKED ISSUE #6008: Change RequestParam to QueryParam for in api-guide.adoc
With the boot 3.x the tests that were changed to use queryParameters instead of requestParameters will produce `query-parameters.adoc` vs `request-parameters.adoc`.   This causes `Unresolved directive in api-guide.adoc` errors in the generated docs.   

For all tests in docs that switched to use `queryParameters` will need their associated entry in `api-guide.adoc` updated to query-parameters.adoc and changed the associated header as well from `Request Parameters` to `Query Parameters`

### PR DESCRIPTION
**Pull request by corneil**
Update request-parameters to query-parameters
Added optional where applicable
Added snippet template to query-parameters and updated template for path-parameters to improve readability.

Fixes #6008

### CODE REVIEW

**Review by onobc — APPROVED** [review]
Thanks for this improvement @corneil . Nice work!

File: spring-cloud-dataflow-classic-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/TaskSchedulerDocumentation.java [hunk 1]
```diff
@@ -121,10 +121,10 @@ void listAllSchedules() throws Exception {
 				.andExpect(status().isOk())
 				.andDo(this.documentationHandler.document(
 						queryParameters(
-								parameterWithName("page")
-										.description("The zero-based page number (optional)"),
+								parameterWithName("page").optional()
+										.description("The zero-based page number"),
 								parameterWithName("size")
-										.description("The requested page size (optional)")),
+										.description("The requested page size")),
```
Comment (cppwfs) [inline spring-cloud-dataflow-classic-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/TaskSchedulerDocumentation.java:127]: We are missing the `.optional()` method here.

**Review by cppwfs — CHANGES_REQUESTED** [review]

**Comment by cppwfs** [comment]
@corneil Looks really good.  Just a one small correction, one question, and a nitpick

File: spring-cloud-dataflow-classic-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/TaskSchedulerDocumentation.java (hunk 1 above)
Comment (corneil) [reply spring-cloud-dataflow-classic-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/TaskSchedulerDocumentation.java:127]: Thanks. Fixed

File: spring-cloud-dataflow-classic-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/TaskExecutionsDocumentation.java [hunk 2]
```diff
@@ -278,15 +281,19 @@ void listTaskThinExecutions() throws Exception {
 
 		this.mockMvc.perform(
 				get("/tasks/thinexecutions")
-					.queryParam("page", "1")
-					.queryParam("size", "2"))
+					.queryParam("page", "0")
+					.queryParam("size", "10")
+					.queryParam("sort", "END_TIME,desc")
+				)
 			.andDo(print())
```
[the remark this answers is not in the document]
Comment (corneil) [reply spring-cloud-dataflow-classic-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/TaskExecutionsDocumentation.java:288]: Removed print from all documentation classes.

**Review by cppwfs — APPROVED** [review]

**Comment by cppwfs** [comment]
@corneil thanks for the good work and investigation on this!

File: spring-cloud-dataflow-classic-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/TaskExecutionsDocumentation.java [hunk 3]
```diff
@@ -233,15 +233,18 @@ void listTaskExecutions() throws Exception {
 
 		this.mockMvc.perform(
 						get("/tasks/executions")
-								.queryParam("page", "1")
-								.queryParam("size", "2"))
+								.queryParam("page", "0")
```
[the remark this answers is not in the document]
Comment (corneil) [reply spring-cloud-dataflow-classic-docs/src/test/java/org/springframework/cloud/dataflow/server/rest/documentation/TaskExecutionsDocumentation.java:236]: To ensure result has data we needed page=0.
I thought size=10 is a good consistent value across usages.

**Review by cppwfs — APPROVED** [review]
@corneil thanks for the good work and investigation on this!

**Pull request merged**